content/api/configuration.md

```diff
@@ -1,10 +1,1407 @@
+  exprContextRegExp: /^\.\/.*$/,
+  exprContextRecursive: true,
+  exprContextCritical: true,
+  wrappedContextRegExp: /.*/,
+  wrappedContextRecursive: true,
+  wrappedContextCritical: false
+}
+```
+
+?> TODO: need help on this, frankly I haven't needed these options and am a bit confused about what they're used for even after reading through the [current section](http://webpack.github.io/docs/configuration.html#automatically-created-contexts-defaults-module-xxxcontextxxx) a few times.
+
+---
+
+### `plugins`
+
+`array`
+
+A list of [webpack plugins]() used for a wide variety of tasks. For example, when multiple bundles share some of the same dependencies the [CommonsChunkPlugin]() could be useful to extract these into a shared bundle. This could be added like so:
+
+```js
+plugins: [
+  new webpack.optimize.CommonsChunkPlugin({
+    ...
+  })
+]
+```
+
+### `resolve`
+
+`object`
+
+Configure how modules are resolved. For example, when calling `import "lodash"` in ES6, the `resolve` options can change where webpack goes to look for `"lodash"` (see [modulesDirectories]()).
+
+---
+
+### `resolve.root`
+
+`string` `array`
+
+Tell webpack what directories should be searched when resolving modules. 
+
+```js
+root: __dirname + '/src'
+```
+
+W> The value or values **must be an absolute path(s)**.
+
+---
+
+### `resolve.fallback`
+
+`string` `array`
+
+Add a fallback(s) for instances where webpack is unable to resolve a module in the given `root` or `modulesDirectories`. This option takes the same values as `root` above.
+
+W> As with `root`, the value or values **must be an absolute path(s)**.
+
+---
+
+### `resolve.modulesDirectories`
+
+`array`
+
+Determine what directories should be searched for installed packages and libraries. These directories will be scanned for similarly to how Node scans for `node_modules`... by looking through the current directory as well as it's ancestors (i.e. `./node_modules`, `../node_modules`, and on). It defaults to:
+
+```js
+modulesDirectories: [ "node_modules", "web_modules" ]
+```
+
+Unlike `root` and `fallback`, **absolute paths are not necessary** and should only be used when there is a hierarchy within these folders.
+
+---
+
+### `resolve.extensions`
+
+`array`
+
+Automatically resolve certain extensions. This defaults to:
+
+```js
+extensions: [ "", ".webpack.js", ".web.js", ".js" ]
+```
+
+which is what enables users to leave off the extension when importing:
+
+```js
+import File from '../path/to/file'
+```
+
+W> Setting this option will **override the default array**, meaning that webpack will no longer try to resolve modules using the default extensions. For modules that are imported with their extension, e.g. `import SomeFile from "./somefile.ext"`, to be properly resolved, an empty string must be included in the array. Similarly, for modules that are imported without extensions, e.g. `import _ from "underscore"`, to be resolved to files with `.js` extensions, you must include ".js" in your array.
```

Comment:
Yea I'll work on it...

> Similarly, for modules that are imported without extensions, e.g. `import _ from "underscore"`, to be resolved to files with `.js` extensions, you must include ".js" in your array.

I've ran into this before and it had me pulling my hair out for a little which is why I felt the need to mention here. However I feel like I read somewhere that you can't override the empty string anymore in version 2? Is that true? If so, maybe it can just be removed.